vitest.workspace.ts

```diff
@@ -0,0 +1,36 @@
+import { defineWorkspace } from 'vitest/config';
+
+export default defineWorkspace([
+  {
+    test: {
+      name: 'node',
+      include: ['./test/node/**/*.test.ts'],
+      environment: 'node',
+    },
+  },
+  {
+    test: {
+      name: 'jsdom',
+      include: ['./test/dom/**/*.test.ts'],
+      environment: 'jsdom',
+    },
+  },
+  {
+    test: {
+      name: 'happy-dom',
```

Comment:
I love Vitest 💚. Now we have tests not only for no DOM, but jsdom, happy-dom, _AND_ chromium via Playwright.

We _can_ also run the same tests in all environments, but I did happen to split them up because I found the setup easier to split. Can possibly recombine in the future